core/src/main/java/org/apache/gravitino/connector/credential/SupportsCredentialOperations.java

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.gravitino.connector.credential;
+
+import java.util.List;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.credential.Credential;
+
+/**
+ * This interface defines how to get credentials, the catalog operation must implement this
+ * interface to support credential vending.
+ */
+public interface SupportsCredentialOperations {
```

Comment:
Why do we need two `SupportsXXX` interfaces, one for `BaseCatalog`, one for `CatalogOperations`? It's hard to understand, also confuses the users.